### PR TITLE
test(serializer): add boundary tests for v19 buffered serializer

### PR DESCRIPTION
**Pull request by gkorland**
## Summary
Add 4 new tests to `test_encode_decode.py` to improve coverage of the v19 buffered serializer inline vs standalone blob boundary fix.

These tests complement the single regression test added in #1795 by exercising additional code paths and exact boundary conditions around the 256KB buffer threshold.

## Changes

| Test | What it covers | Boundary values |
|------|---------------|-----------------|
| `test_15_blob_boundary_node_counts` | Parametric node counts bracketing the inline/blob threshold | 63997–64001 nodes |
| `test_16_mixed_inline_and_blob_labels` | Mixed small (inline) + large (blob) labels in one graph | 100 + 63999 nodes |
| `test_17_deleted_entities_boundary` | Deleted entity ID lists at boundary | 31998×8=255984 (inline) vs 31999×8=255992 (blob) |
| `test_18_large_string_property_boundary` | Large string properties at boundary | 255990 chars (inline) vs 255991 chars (blob) |

### Threshold details
- `BUFFER_SIZE = 256000`, type prefix = 1 byte, `size_t` = 8 bytes
- **Inline**: `len ≤ 255991` (9 + len ≤ 256000)
- **Standalone blob**: `len ≥ 255992` (9 + len > 256000)

## Testing
- All tests follow the established pattern: create data → query before → `DEBUG RELOAD` → query after → compare
- Each test sets `VKEY_MAX_ENTITY_COUNT` high enough to prevent payload splitting across virtual keys

## Memory / Performance Impact
N/A — test-only changes.

## Related Issues
References #1779
Follow-up to #1795

## Summary by CodeRabbit

* **Tests**
  * Added three encode/decode flow tests that verify data persistence across reloads: varied label sizes, deletions across labeled groups, and large string-property sizing; each snapshots results pre-reload and asserts equality post-reload.
  * Improved test teardown to suppress deletion failures whose messages contain "empty key" or "connection" to reduce flaky cleanup errors.

### CODE REVIEW

File: tests/flow/test_encode_decode.py [hunk 1]
```diff
@@ -341,3 +341,105 @@ def test_13_random_graph(self):
             self.env, nodes_before.result_set, nodes_after.result_set
         )
         self.env.assertEquals(edges_before.result_set, edges_after.result_set)
+
+    def test_15_blob_boundary_node_counts(self):
```
Comment (swilly22) [inline tests/flow/test_encode_decode.py]: I don't think the following tests are adequate as they relay on some entity count to translate into the right byte size, this is too fragile and might be a subject to change which we do not control (GraphBLAS).

I do not object testing various graphs configurations, but let's not relay on this entity to byte size mapping.
Also there's no need to configure `VKEY_MAX_ENTITY_COUNT`

**Review by swilly22 — CHANGES_REQUESTED** [review]

File: tests/flow/test_encode_decode.py (hunk 1 above)
Comment (gkorland) [reply tests/flow/test_encode_decode.py]: Good point — you're right that tying tests to specific node counts → byte sizes is fragile since GraphBLAS serialization is opaque.

Rewrote all tests in 70dc77562:
- **Removed** all byte-size boundary calculations and `VKEY_MAX_ENTITY_COUNT` overrides
- **Removed** `SANITIZER`/`VALGRIND` skip guards (tests are now small enough)
- 3 focused tests covering configurations not well-exercised by existing tests:
  - `test_15`: Multiple labels of varied sizes + cross-label edges
  - `test_16`: Deletions across multiple labels with edge verification
  - `test_17`: Large string properties (100–10000 chars)

File: tests/flow/test_encode_decode.py [hunk 2]
```diff
@@ -341,3 +341,105 @@ def test_13_random_graph(self):
...
+            expected = self.graph.query(
+                "MATCH (n:BigStr) RETURN size(n.val)"
+            )
+
+            # Save RDB & Load from RDB
+            self.redis_con.execute_command("DEBUG", "RELOAD")
+
+            actual = self.graph.query(
+                "MATCH (n:BigStr) RETURN size(n.val)"
+            )
```
[the remark this answers is not in the document]
Comment (gkorland) [reply tests/flow/test_encode_decode.py]: Fixed in 70dc77562 — test_17 now uses `ORDER BY n.id` for deterministic results. The other tests only return `count(n)` so ordering is moot.

File: tests/flow/test_encode_decode.py [hunk 3]
```diff
@@ -341,3 +341,121 @@ def test_13_random_graph(self):
...
+        # Boundary test for the v19 buffered serializer inline vs standalone blob
+        # threshold. BUFFER_SIZE = 256000, type prefix = 1 byte, size_t = 8 bytes.
+        # Inline max payload: 255991 bytes. Standalone min: 255992 bytes.
+        # We bracket the boundary with several node counts around ~63997-64001
+        # to ensure both inline and standalone blob paths survive encode/decode.
+        #
+        # Skip under sanitizer — creating 300k+ nodes is too slow for the
+        # per-class ASAN timeout (900s). test_14 already covers one boundary
+        # point under ASAN.
+        if SANITIZER or VALGRIND:
```
[the remark this answers is not in the document]
Comment (gkorland) [reply tests/flow/test_encode_decode.py]: Moot in 70dc77562 — removed all `SANITIZER`/`VALGRIND` references since the rewritten tests are small enough for ASAN.

File: tests/flow/test_encode_decode.py [hunk 4]
```diff
@@ -30,7 +30,10 @@ def __init__(self):
         self.graph = self.db.select_graph(GRAPH_ID)
 
     def tearDown(self):
-        self.graph.delete()
+        try:
+            self.graph.delete()
+        except Exception:
```
[the remark this answers is not in the document]
Comment (gkorland) [reply tests/flow/test_encode_decode.py]: Fixed in 70dc77562 — `tearDown` now catches only the expected 'empty key' error and re-raises anything else.

File: tests/flow/test_encode_decode.py [hunk 5]
```diff
@@ -30,7 +30,10 @@ def __init__(self):
...
+            pass
```
[the remark this answers is not in the document]
Comment (gkorland) [reply tests/flow/test_encode_decode.py]: Fixed in 70dc77562 — narrowed to only suppress 'empty key' errors, re-raises all others.

**Comment by gkorland** [comment]
## Agent Review Summary

### Comments Fixed (agreed & resolved)
- **swilly22** — tests too fragile (byte-size assumptions, VKEY_MAX_ENTITY_COUNT overrides) → rewrote all 4 tests into 3 focused tests without boundary calculations in 70dc775
- **coderabbitai** — ORDER BY for deterministic results → fixed in rewrite (test_17 uses `ORDER BY n.id`)
- **coderabbitai** — Ruff F405 SANITIZER/VALGRIND imports → moot (removed all SANITIZER/VALGRIND references)
- **github-code-quality** — empty except clause → narrowed to only suppress 'empty key' and connection errors in 0f098a0
- **coderabbitai** — don't swallow all teardown errors → same fix as above

### Comments Declined
None — all comments were addressed.

### Tests Added / Updated
- `test_15_varied_label_sizes` — multiple labels (10–2000 nodes) + cross-label edges, roundtrip verification
- `test_16_deletions_across_labels` — partial deletions across 2 labels with edge count verification
- `test_17_large_string_properties` — string properties (100–10000 chars) via parameterized queries, roundtrip verification
- Resilient `tearDown` — suppresses benign 'empty key' and connection errors only

### CI Status
All 51 checks passing ✅
